examples/rest-api/README.md

```diff
@@ -0,0 +1,58 @@
+# Zync REST API example
+
+This example project implements Zync's REST API protocol to synchronize OAuth2 clients.
+
+## Prerequisites
+
+Given 3scale API is configured to use:
+  * OpenID Connect as the Authentication,
+  * "REST API" as a OpenID Connect Issuer Type and
+  * "http://id:secret@example.com/api" as OpenID Connect Issuer.
+
+When a 3scale application is created/updated/deleted zync will try to replay that change to "http://example.com/api".
+
+## Creating, updating and deleting Clients
+
+Zync will make following requests to create/update/delete clients:
+
+* `PUT /clients/:client_id` (create, update)
+* `DELETE /clients/:client_id` (delete)
+
+All endpoints must reply 2xx status code. Otherwise the request will be retried.
```

Comment:
@mikz is there any limit on number of period of retries? 

What if the DELETE responds with `404`? I believe it can cause an infinite loop of trying to delete the non-existing client, e.g. https://issues.jboss.org/browse/THREESCALE-2206

Reply:
Then it should respond with 200 that the client is gone. It has exponential backoff and yes, there is some limit. It is the same for every object.

Reply:
@mikz Can you please point me to the place where this is configured/set?

Reply:
I believe that for most tasks we are using the default https://github.com/chanks/que/blob/master/docs/error_handling.md#error-handling

Reply:
@mikz thanks for the info!